File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-2.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def __init__(self):
        self.res = None

    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:
        self.helper(root, p.val, q.val)
        return self.res
        
    def helper(self, node, p, q):
        ancestor = False
        if not node:
            return False
        ret1 = self.helper(node.left, p, q)
        ret2 = self.helper(node.right, p, q)
        if node.val == p or node.val == q:
            ancestor = True
        if ancestor and (ret1 or ret2):
            self.res = node
        if ret1 and ret2:
            self.res = node
        return ancestor or ret1 or ret2

        
        
        

File: Data Structures & Algorithms/invert-a-binary-tree/submission-2.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {TreeNode}
     */
    invertTree(root) {
        function helper(node){
            if (node){
                helper(node.right)
                helper(node.left)
                let temp = node.right
                node.right = node.left
                node.left = temp
            }
            return
        }
        if (root) {
            helper(root);
            return root;
        } else return null
    }
}
